src/coreclr/src/pal/src/exception/machexception.cpp

```diff
@@ -1319,15 +1510,29 @@ InjectActivationInternal(CPalThread* pThread)
                 *(--sp) = ThreadState.__rip;
                 *(--sp) = ThreadState.__rbp;
                 size_t rbpAddress = (size_t)sp;
+#elif defined(HOST_ARM64)
+            if ((g_safeActivationCheckFunction != NULL) && g_safeActivationCheckFunction(arm_thread_state64_get_lr(ThreadState), /* checkingCurrentThread */ FALSE))
```

Comment:
We should check PC here, not LR. We are interested in the current execution point of the thread, not its return address.